corehq/apps/users/tests/test_device_rate_limiter.py

```diff
@@ -1,27 +1,30 @@
-from django.test import SimpleTestCase, override_settings
+from django.test import TestCase
 from freezegun import freeze_time
 
 from corehq.apps.cloudcare.const import DEVICE_ID as CLOUDCARE_DEVICE_ID
-from corehq.apps.users.device_rate_limiter import device_rate_limiter
-from corehq.tests.pytest_plugins.reusedb import clear_redis
-from corehq.util.test_utils import flag_enabled
+from corehq.apps.users.device_rate_limiter import DEVICE_LIMIT_PER_USER_KEY, REDIS_KEY_PREFIX, device_rate_limiter
+from corehq.project_limits.models import SystemLimit
+from corehq.util.test_utils import flag_disabled, flag_enabled
 
 
 @freeze_time("2024-12-10 12:05:43")
-@override_settings(DEVICE_LIMIT_PER_USER=1)
-@override_settings(ENABLE_DEVICE_RATE_LIMITER=True)
-class TestDeviceRateLimiter(SimpleTestCase):
+@flag_enabled("DEVICE_RATE_LIMITER")
+class TestDeviceRateLimiter(TestCase):
 
     domain = 'device-rate-limit-test'
 
     def setUp(self):
-        self.addCleanup(clear_redis)
+        SystemLimit.objects.create(key=DEVICE_LIMIT_PER_USER_KEY, limit=1)
+
+    def tearDown(self):
+        for key in device_rate_limiter.client.scan_iter(f"{REDIS_KEY_PREFIX}*"):
+            device_rate_limiter.client.delete(key.decode('utf-8'))
```

Comment:
Is this still needed since the global cache patch has been added?

Reply:
Yeah the global cache patch is for SystemLimit, which is just used by the device rate limiter, but the rate limiter still stores its relevant info in redis and needs to clear it after test runs.